plugin/src/main/java/org/opensearch/ml/indices/MLIndicesHandler.java

```diff
@@ -87,19 +89,24 @@ public void initMLTaskIndex(ActionListener<Boolean> listener) {
 
     public void initMLIndexIfAbsent(String indexName, String mapping, ActionListener<Boolean> listener) {
         if (!clusterService.state().metadata().hasIndex(indexName)) {
-            CreateIndexRequest request = new CreateIndexRequest(indexName).mapping("_doc", mapping, XContentType.JSON);
-
-            client.admin().indices().create(request, ActionListener.wrap(r -> {
-                if (r.isAcknowledged()) {
-                    log.info("create index:{}", indexName);
-                    listener.onResponse(true);
-                } else {
-                    listener.onResponse(false);
-                }
-            }, e -> {
-                log.error("Failed to create index " + indexName, e);
+            try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+                ActionListener<CreateIndexResponse> actionListener = ActionListener.wrap(r -> {
+                    if (r.isAcknowledged()) {
+                        log.info("create index:{}", indexName);
+                        listener.onResponse(true);
+                    } else {
+                        listener.onResponse(false);
+                    }
+                }, e -> {
+                    log.error("Failed to create index " + indexName, e);
+                    listener.onFailure(e);
+                });
+                CreateIndexRequest request = new CreateIndexRequest(indexName).mapping("_doc", mapping, XContentType.JSON);
+                client.admin().indices().create(request, ActionListener.runBefore(actionListener, () -> threadContext.restore()));
```

Comment:
If we are running before creation `threadContext.restore()` doesn't that mean we restore original thread context and user info is restored so we can't create system index or is this fine cause the CreateIndexRequest itself is one line before?

Reply:
The execution order of the code
1. create admin `client.admin().indices().create`
2. restore thread context `threadContext.restore()` 
3. run actionListener

So restore original context happens after index created.